include/ck_tile/ops/gemm/kernel/grouped_gemm_kernel.hpp

```diff
@@ -0,0 +1,368 @@
+                throw std::runtime_error("wrong! group_count_ != p_As/b/c.size");
+            }
+            gemm_kernel_args_.reserve(group_count_);
+
+            for(std::size_t i = 0; i < gemm_descs.size(); ++i)
+            {
+                const index_t M = gemm_descs[i].M;
+                const index_t N = gemm_descs[i].N;
+                const index_t K = gemm_descs[i].K;
+
+                if(M == 0)
+                {
+                    continue;
+                }
+
+                const index_t stride_a = gemm_descs[i].stride_A;
+                const index_t stride_b = gemm_descs[i].stride_B;
+                const index_t stride_c = gemm_descs[i].stride_C;
+
+                const index_t MPerBlock = TilePartitioner::MPerBlock;
+                const index_t NPerBlock = TilePartitioner::NPerBlock;
+
+                const auto M0 = ck_tile::integer_divide_ceil(M, MPerBlock);
+                const auto N0 = ck_tile::integer_divide_ceil(N, NPerBlock);
+
+                const index_t grid_size_one = M0 * N0 * 1;
+
+                const index_t block_start = grid_size_;
+                const index_t block_end   = grid_size_ + grid_size_one;
+
+                grid_size_ += grid_size_one;
+
+                auto karg = GroupedKernelArgument{type_convert<const ADataType*>(p_As[i]),
+                                                  type_convert<const BDataType*>(p_Bs[i]),
+                                                  type_convert<CDataType*>(p_Es[i]),
+                                                  M,
+                                                  N,
+                                                  K,
+                                                  stride_a,
+                                                  stride_b,
+                                                  stride_c};
+
+                gemm_kernel_args_.emplace_back(
+                    std::move(karg), block_start, block_end, group_count_, N0);
+            }
+        }
+
+        std::vector<GemmTransKernelArg> gemm_kernel_args_;
+        index_t grid_size_;
+        void* p_workspace_ = nullptr;
+    };
+
+    __host__ static constexpr auto GridSize(const Argument& args)
+    {
+        return dim3(args.grid_size_, 1, 1);
+    }
+
+    CK_TILE_HOST static auto MakeKargs(std::vector<const void*>& p_As,
+                                       std::vector<const void*>& p_Bs,
+                                       std::vector<void*>& p_Es,
+                                       const std::vector<GroupedGemmDesc>& gemm_descs)
+    {
+        return Argument{p_As, p_Bs, p_Es, gemm_descs};
+    }
+
+    CK_TILE_HOST_DEVICE static constexpr index_t GetSmemSize()
+    {
+        return max(GemmPipeline::GetSmemSize(), EpiloguePipeline::GetSmemSize());
+    }
+
+    CK_TILE_DEVICE void Run(const GroupedKernelArgument& kargs,
+                            const index_t block_start,
+                            const index_t block_n0_size) const
+    {
```

Comment:
This function should take here this structure: https://github.com/ROCm/composable_kernel/blob/develop/include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp#L38 and TilePartitioner object.